addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -134,10 +134,11 @@ odoo.define('website.s_website_form', function (require) {
                     var $field = self.$target.find('input[name="' + field + '"], textarea[name="' + field + '"]');
                     if (!$field.val() && _.has(dataForValues, field) && dataForValues[field]) {
                         $field.val(dataForValues[field]);
+                        $field.data('website_form_original_default_value', $field.val());
                     } else if (!$field.val() && $field.data('fill-with')) {
                         $field.val(self.preFillValues[$field.data('fill-with')] || '');
+                        $field.data('website_form_original_default_value', $field.val());
```

Comment:
The piece of code you point out is useful, it was added [here](https://github.com/odoo/odoo/pull/50745/commits/48475eb6d879c885baa14bfee3130add019d6418) so that when you edit a form to apply for a job, you don't save the job_id as a default value. This job_id must be different for each job. So we have to delete its value => maybe it can be done differently but we can't just delete this piece of code.

This piece of code deletes all the values of the fields which have the property `website_form_original_default_value`. My solution is to put this property only when it is relevant.

Reply:
Neither the explanation nor the fix you mention seems good. As as explained above, the code [here](https://github.com/odoo/odoo/pull/50745/commits/48475eb6d879c885baa14bfee3130add019d6418#diff-e712b94d96fb636dd30632e9540fa8caa06ac145c2618544007b5f29ba77b364R66) has no impact on the DOM. So [doing that](https://github.com/odoo/odoo/pull/50745/commits/48475eb6d879c885baa14bfee3130add019d6418#diff-ec70bde754e307ff6e9c3f50d61e703606fc40619e066464c99ffaddfdbb90aeR442) has simply no effect (again, as I explained above). So if NLE made this code I guess it solves something but I'll need more explanation to understand why. And I really doubt this was the right fix, even by chance. Can you investigate and explain to me why what I explained above is wrong?

Reply:
Ok I investigated and here is the point: when you do `.val('XXX')` on a `input` with `type="hidden"` then the value attribute will be set to _XXX_. (That's the case for the job_id stuff)
You can check it [here](https://jsfiddle.net/gdi_odoo/f1j7apk8/1/)

Reply:
Thanks, indeed :+1: That was missing from commit message and code comments for the NLE's script: it was really useful to investiguate. Now can't this be changed ?

A) At best, the public widget (000.js) should not modify the DOM on start (only the value properties like I thought it was doing). For the hidden fields, maybe the solution would be to auto-fill them on submit rather than on page load (that would thus not impact the edition at all).

B) If not possible for some reason (why?), I think NLE's fix has to be reviewed to:
    1. Add more comments explaining why following your investigation
    2. It's always bad to have code in public widgets that is cleaned by the editor. Normally I think you could have done what the cleanForSave does in the public widget's destroy method (thus keeping the code together and probably simplifying it (you can just memorize the inputs there is to clean on start and loop through them on destroy)

Reply:
> 2. It's always bad to have code in public widgets that is cleaned by the editor. Normally I think you could have done what the cleanForSave does in the public widget's destroy method (thus keeping the code together and probably simplifying it (you can just memorize the inputs there is to clean on start and loop through them on destroy)

@Guillaume-gdi What about this? I can't remember you proved me this was wrong? (not needing `cleanForSave` like you explain in your code comment above)